### ISSUE TITLE
Dynamic method chaining possible? 

### ISSUE THREAD

**Issue by joshRpowell** [DmitryTsepelev/store_model#45]
Is it possible to method chain dynamically?  Something like the following? 

```ruby
def self.send_chain(methods)
  methods.inject(self, :send)
end
```

Where I could:
```ruby
> Article.send_chain(["with_author", "pending_review"])
```

**Comment by DmitryTsepelev** [comment]
Hi @joshRpowell! Could you please provide more detailed example? The gem does not work with relations so I'm curios what problem are you trying to solve 🙂

**Comment by joshRpowell** [comment]
@DmitryTsepelev I'll admit including the relation result was misleading (edited for clarity). I had grabbed a quick example I was working with at the end of a long day.  😪 

💥 well...while writing this up, I answered my own question!  `send_chain` as an instance method on the JSON field does the trick.  

```ruby
class MetaInf
  include StoreModel::Model

  attribute :params, Params.to_type

  def send_chain(methods)
    methods.inject(self, :send)
  end
end

c.meta_inf.send_chain([:params, :role_a, :first_name])
=> "HAPPY"
```

**TLDR: my original write-up explaining my issue.**

Here's my store_model set-up: 

```ruby
class Case < ApplicationRecord
  attribute :meta_inf, MetaInf.to_type
end

class MetaInf
  include StoreModel::Model

  attribute :params, Params.to_type
end

class Params
  include StoreModel::Model

  attribute :role_a, RoleA.to_type
end

class RoleA
  include StoreModel::Model

  attribute :last_name, :string
  attribute :first_name, :string
  attribute :middle_name, :string
  attribute :date_of_birth, :string
end
```

And I can do the following: 
```ruby
c = Case.take
=> #<Case:0x00007fb28098aff0
 id: 670,
 meta_inf:   #<MetaInf params: #<Params:0x00007fb2802e3c38>,
 created_at: Thu, 25 Oct 2018 15:27:11 UTC +00:00,
 updated_at: Tue, 05 Nov 2019 22:02:31 UTC +00:00>

c.meta_inf.class 
=> MetaInf

c.meta_inf.params
=> #<Params petitioner: nil, role_a: #<RoleA:0x00007fb2803c84f0> >

c.meta_inf.params.role_a
=> #<RoleA last_name: PERSON, first_name: HAPPY, middle_name: nil, date_of_birth: 19731008>

c.meta_inf.params.role_a.first_name
=> "HAPPY"
```

As the JSON grows, in this example, multiple roles could be added. In which case i would want to query the following:

```ruby
c.meta_inf.params.role_a.first_name
...
c.meta_inf.params.role_z.first_name
```

I'd like to use dynamically define the path the first name in this simple example (actual use case more complex).
```ruby
def self.send_chain(methods)
  methods.inject(self, :send)
end

QUERY_PARAMS = [
  [:params, :role_a, :first_name],
  ...
  [:params, :role_z, :first_name]
]

list_of_first_names = QUERY_PARAMS.map do |query|
  c.meta_inf.send_chain(query)  
  # equivalent to c.meta_inf.send(:params).send(:role_a).send(:first_name)
  # or c.meta_inf.params.role_a.first_name
end
```

**Comment by DmitryTsepelev** [comment]
Yep, got it, your solution should work for this complex cases. Also, for some simpler situations good old [Array#dig](https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig) can be used too.
